packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3452,3 +3478,49 @@ const waitForSeq = async (
 			deltaManager.on("op", handleOp);
 		}
 	});
+
+class OpGroupingManager {
+	static groupedBatchOp = "groupedBatch";
+
+	public submitBatch(
+		context: IContainerContext,
+		batch: IBatchMessage[],
+		referenceSequenceNumber?: number | undefined,
+	): number {
+		if (batch.length < 2) {
+			return context.submitBatchFn(batch, referenceSequenceNumber);
+		}
+		const groupedOp: IBatchMessage = {
+			metadata: undefined,
+			compression: batch[0].compression,
```

Comment:
we are encoding here a dependency that only first op will have compression property. If it were to change, this code would break things, so I think it would be nice to assert in variants (i.e. assert that all but first message do not have compression bits).
That said, compression bit on op envelop was useful to let service know that it can recompress content if it choses. so.
With chunking and grouping of ops this capability is lost (service will not have enough knowledge how to parse these messages). So it feels to me compression should not be on groupedOp, instead it should be part of content (and I think we already implicitly do so by taking whole `batch` as content)